src/diagram/models/mathjs-utils.ts

```diff
@@ -50,12 +40,21 @@ export const replaceInputNames = (expression: string, inputNames: (string | unde
       inputNames.forEach((name, index) => {
         if (symbol.name === name) {
           symbol.name = `input_${index}`;
+          inputsInExpression.push(name);
         }
       });
     }
-    return expressionNode.toString();
+    return { expression: expressionNode.toString(), inputsInExpression };
   } catch (e) {
-    // if there is parse error just return the expression for now
-    return localExpression;
+    // if there is parse error, return the original expression for now
+    return { expression: localExpression, inputsInExpression };
   }
 };
+
+export const replaceInputNames = (expression: string, inputNames: (string | undefined)[]) => {
+  return parseExpression(expression, inputNames).expression;
+};
+
+export const getUsedInputs = (expression: string, inputNames: (string | undefined)[]) => {
+  return parseExpression(expression, inputNames).inputsInExpression;
+};
```

Comment:
Does this seem pretty straightforward? I didn't want to duplicate a lot of the original `replaceInputNames` function's code, but I'm not sure this is the clearest way to do that.

Reply:
Yes this looks good.